testutil/integration/router.go

```diff
@@ -63,8 +64,8 @@ func NewIntegrationApp(
 
 	bApp.SetInitChainer(func(ctx sdk.Context, _ *cmtabcitypes.RequestInitChain) (*cmtabcitypes.ResponseInitChain, error) {
 		for _, mod := range modules {
-			if m, ok := mod.(module.HasGenesis); ok {
-				m.InitGenesis(ctx, appCodec, m.DefaultGenesis(appCodec))
+			if m, ok := mod.(appmodulev2.HasGenesis); ok {
```

Comment:
nit, as it works with baseapp I would use appmodule.HasGenesis here instead and avoid using appmodulev2 for baseapp related things.